crates/router/src/lib.rs

```diff
@@ -88,36 +89,100 @@ pub fn mk_app(
 
     #[cfg(feature = "stripe")]
     {
         server_app = server_app.service(routes::StripeApis::server(state.clone()));
     }
-    server_app = server_app.service(routes::Health::server(state));
+    server_app = server_app.service(routes::Health::olap_server(state));
     server_app
 }
 
 #[allow(clippy::expect_used, clippy::unwrap_used)]
 /// # Panics
 ///
 ///  Unwrap used because without the value we can't start the server
-pub async fn start_server(conf: Settings) -> ApplicationResult<(Server, AppState)> {
+#[cfg(feature = "olap")]
+pub async fn start_olap_server(conf: settings::Settings) -> ApplicationResult<(Server, AppState)> {
     logger::debug!(startup_config=?conf);
     let server = conf.server.clone();
     let state = routes::AppState::new(conf).await;
     // Cloning to close connections before shutdown
     let app_state = state.clone();
     let request_body_limit = server.request_body_limit;
-    let server = actix_web::HttpServer::new(move || mk_app(state.clone(), request_body_limit))
+    let server = actix_web::HttpServer::new(move || mk_olap_app(state.clone(), request_body_limit))
+        .bind((server.host.as_str(), server.port))?
+        .workers(server.workers.unwrap_or_else(num_cpus::get_physical))
+        .run();
+
+    Ok((server, app_state))
+}
+
+pub fn mk_oltp_app(
+    state: AppState,
+    request_body_limit: usize,
+) -> actix_web::App<
+    impl ServiceFactory<
+        ServiceRequest,
+        Config = (),
+        Response = actix_web::dev::ServiceResponse<impl MessageBody>,
+        Error = actix_web::Error,
+        InitError = (),
+    >,
+> {
+    let json_cfg = actix_web::web::JsonConfig::default()
+        .limit(request_body_limit)
+        .content_type_required(true)
+        .content_type(|mime| mime == mime::APPLICATION_JSON) // FIXME: This doesn't seem to be enforced.
+        .error_handler(utils::error_parser::custom_json_error_handler);
+
+    let mut server_app = actix_web::App::new()
+        .app_data(json_cfg)
+        .wrap(middleware::RequestId)
+        .wrap(router_env::tracing_actix_web::TracingLogger::default())
+        .wrap(ErrorHandlers::new().handler(
+            StatusCode::NOT_FOUND,
+            errors::error_handlers::custom_error_handlers,
+        ))
+        .wrap(ErrorHandlers::new().handler(
+            StatusCode::METHOD_NOT_ALLOWED,
+            errors::error_handlers::custom_error_handlers,
+        ))
+        .wrap(cors::cors())
+        .service(routes::Payments::oltp_server(state.clone()))
+        .service(routes::Customers::oltp_server(state.clone()))
+        .service(routes::Refunds::oltp_server(state.clone()))
+        .service(routes::Payouts::oltp_server(state.clone()))
+        .service(routes::PaymentMethods::oltp_server(state.clone()))
+        .service(routes::EphemeralKey::oltp_server(state.clone()))
+        .service(routes::Webhooks::oltp_server(state.clone()));
```

Comment:
There seems to be a significant amount of shared code between `mk_olap_app()` and `mk_oltp_app()`, specifically till before services are registered. Can we move that to a separate function that returns the application builder?

Reply:
Done